### PR TITLE
ADR 239 - Avatar assembly definition

### PR DESCRIPTION
**Pull request by davidejensen**
Based on issue #239 this is the ADR about the avatar assembly definition changes

### CODE REVIEW

File: content/ADR-239-avatar-assembly-changes.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+         "hasConnectedWeb3":true
+      }
+   ]
+}
+```
+
+### Avatar assembly suggested flow
+
+With the points previously discovered, the suggested procedure to visualize an avatar is the following:
+* Request a user profile ( `https://catalyst/lambdas/profile/address` )
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:194]: The lambdas endpoint is deprecated
```suggestion
* Request a user profile ( `POST https://content-server/content/active {"pointers": ["0xaddress"]}` )
```

File: content/ADR-239-avatar-assembly-changes.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+* Request the wearables listed in the profile wearables field, all wearables listed are the equipped ones, no matter their visibility. ( `https://catalyst/lambdas/collections/wearables?wearableId=urnId` )
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:195]: The lambdas endpoint is deprecated

```suggestion
* Request the wearables listed in the profile wearables field, all wearables listed are the equipped ones, no matter their visibility. ( `POST https://content-server/content/active {"pointers": ["urn1", "urn2"]}` )
```

File: content/ADR-239-avatar-assembly-changes.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+In order to store the force render categories, it is proposed to add a new field to the player profile holding a list of overridden hide categories.
+This list of string will hold the current overridden categories so that each player can render other players (and itself) according to the new rules.
+As an example:
+```
+"forceRender": ["mask","top_head","feet"]
+```
+This example means that when the avatar is going to be assembled the "mask", "top_head" and "feet" categories will be shown even if other wearables are hiding them.
+
+Here's the avatar definition with the additional forceRender field that will be provided:
+```
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:86]:
````suggestion
```json
````

File: content/ADR-239-avatar-assembly-changes.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+### Force Render
+
+To extend further players ability to use the wearables they have in all their glory. If an equipped wearable is hidden by the effect of another wearable, players can unhide them and override this instruction.
+This forced render effect is removed as soon as another wearable that re-hides that category is equipped, forcing the user to show it again.
+
+In order to store the force render categories, it is proposed to add a new field to the player profile holding a list of overridden hide categories.
+This list of string will hold the current overridden categories so that each player can render other players (and itself) according to the new rules.
+As an example:
+```
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:80]:
````suggestion
```yaml
````

File: content/ADR-239-avatar-assembly-changes.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
+---
+adr: 239
+date: 2023-06-01
+title: Avatar Assembly Changes
+status: Review
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:5]: This PR has a tide effect of potentially dozens/hundreds of hours across many teams.

Please update https://adr.decentraland.org/adr/ADR-57 (Avatar assembling instructions for visualization) to reflect the new changes in this same PR. That way we can use the diff in the previous one to update the:
- bevy implementation https://github.com/decentraland/bevy-explorer/pull/19/files
- babylon implementation https://github.com/decentraland/hammurabi/pull/31
- werable preview for marketplace & for third parties (DG mobile) https://wearable-preview.decentraland.org/?profile=default1
- docs site preview https://docs.decentraland.org/creator/emotes/emotes-overview/

**Review by menduz — CHANGES_REQUESTED** [review]

File: content/ADR-239-avatar-assembly-changes.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+On the base of that the replacing functionality for new wearables will be removed, pre-existing wearables will see the replacing list being merged with the hiding list.
+
+### Body fallback
+
+Previously, by unequipping the upper_body, the lower_body, the feet or the eyebrows, caused the system to fallback to default wearables (t-shirt, pants and shoes).
+This prevents, for example, players from going barefoot or without t-shirt.
+
+The system will extent the concept of Body by removing the default fallback wearables.
+These pieces, which can be called "Bare feet", "Naked Top" and "Naked Bottom", are not considered wearables, but rather parts of the base body geometry that are exclusively displayed when no wearables are equipped and there are no other wearables hiding the relevant category.
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:68]: A stronger algorithmic definition of these rules will be needed to implement the logic. What can we infer is only `name.isOneOf(["Bare feet", "Naked Top", "Naked Bottom"])` and that may not be enough

File: content/ADR-239-avatar-assembly-changes.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+The base body geometry and textures will change to make them more suitable for an environment for all audiences.
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:70]: This comment has nothing to do with the ADR and can be omited
```suggestion
```

File: content/ADR-239-avatar-assembly-changes.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+### Force Render
+
+To extend further players ability to use the wearables they have in all their glory. If an equipped wearable is hidden by the effect of another wearable, players can unhide them and override this instruction.
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:74]: Clarity over narrative
```suggestion
To extend further players ability to use the wearables they have. If an equipped wearable is hidden by the effect of another wearable, players can unhide them and override this instruction.
```

File: content/ADR-239-avatar-assembly-changes.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+This forced render effect is removed as soon as another wearable that re-hides that category is equipped, forcing the user to show it again.
+
+In order to store the force render categories, it is proposed to add a new field to the player profile holding a list of overridden hide categories.
+This list of string will hold the current overridden categories so that each player can render other players (and itself) according to the new rules.
+As an example:
+```
+"forceRender": ["mask","top_head","feet"]
```
Comment (menduz) [inline content/ADR-239-avatar-assembly-changes.md:81]: Maybe, if we are still on-time, forcedCategories would be a better self-explanatory name
```suggestion
"forcedCategories": ["mask","top_head","feet"]
```

Comment (davidejensen) [reply content/ADR-239-avatar-assembly-changes.md:80]: sadly too late

File: content/ADR-239-avatar-assembly-changes.md (hunk 5 above)
Comment (davidejensen) [reply content/ADR-239-avatar-assembly-changes.md:5]: I thought old ADRs were supposed to be set in stone once approved, should I modify that ADR in this PR? Or how should I handle this? @menduz

File: content/ADR-239-avatar-assembly-changes.md (hunk 9 above)
Comment (davidejensen) [reply content/ADR-239-avatar-assembly-changes.md:80]: the functionality has already been added to unity renderer, browser interface and profile deploy validation.
Also I think that forceRender is descriptive enough, considering that something hidden is still equipped, so by adding it to the list we are forcing it to render even if hidden

File: content/ADR-239-avatar-assembly-changes.md (hunk 5 above)
Comment (davidejensen) [reply content/ADR-239-avatar-assembly-changes.md:5]: done

**Review by menduz — APPROVED** [review]

**Pull request merged**